### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by anshulv1401**
[MOSIP-25505](https://mosip.atlassian.net/browse/MOSIP-25505) RTL support added
[MOSIP-25770](https://mosip.atlassian.net/browse/MOSIP-25770) info icon on the claim added
[MOSIP-25870](https://mosip.atlassian.net/browse/MOSIP-25870) replacing MOSIP with e-signet
[MOSIP-25767](https://mosip.atlassian.net/browse/MOSIP-25767), [MOSIP-25877](https://mosip.atlassian.net/browse/MOSIP-25877) transaction overriding for 2 login - bug fixed
[MOSIP-25883](https://mosip.atlassian.net/browse/MOSIP-25883) Captcha support in UI
Storing encoded OAuth details in URL and adding hash of the OAuth details in all auth requests.

[MOSIP-25505]: https://mosip.atlassian.net/browse/MOSIP-25505?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MOSIP-25770]: https://mosip.atlassian.net/browse/MOSIP-25770?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[MOSIP-25870]: https://mosip.atlassian.net/browse/MOSIP-25870?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[MOSIP-25767]: https://mosip.atlassian.net/browse/MOSIP-25767?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MOSIP-25877]: https://mosip.atlassian.net/browse/MOSIP-25877?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[MOSIP-25883]: https://mosip.atlassian.net/browse/MOSIP-25883?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by gsasikumar — CHANGES_REQUESTED** [review]

File: .env [hunk 1]
```diff
@@ -27,4 +27,7 @@ REACT_APP_INJI_ENABLE="true"
 #inclusive
 REACT_APP_SBI_PORT_RANGE="4501-4600"
 REACT_APP_RESEND_OTP_TIMEOUT_IN_SEC="120"
-REACT_APP_SEND_OTP_CHANNELS="email,mobile"
\ No newline at end of file
+REACT_APP_SEND_OTP_CHANNELS="email,mobile"
+REACT_APP_SEND_OTP_SHOW_CAPTCHA="false"
+#test site key
+REACT_APP_SEND_OTP_CAPTCHA_SITE_KEY="6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI"
```
Comment (gsasikumar) [inline .env:33]: This is supposed to be secret? do you need a valid one here? or can you have just a dummy value? also revoke this site key.

File: .env [hunk 2]
```diff
@@ -27,4 +27,7 @@ REACT_APP_INJI_ENABLE="true"
 #inclusive
 REACT_APP_SBI_PORT_RANGE="4501-4600"
```
Comment (gsasikumar) [inline .env:28]: Why are we calling everything as REACT_APP ?? This is eSignet correct?

File: src/services/authService.js [hunk 3]
```diff
@@ -153,13 +153,14 @@ class authService {
    * @param {List<string>} otpChannels list of channels(ie. phone, email)
    * @returns /send-otp API response
    */
-  post_SendOtp = async (transactionId, individualId, otpChannels) => {
+  post_SendOtp = async (transactionId, individualId, otpChannels, captchaToken) => {
```
Comment (gsasikumar) [inline src/services/authService.js:156]: What if a captcha is not needed?

File: .env [hunk 4]
```diff
@@ -27,4 +27,7 @@ REACT_APP_INJI_ENABLE="true"
...
 REACT_APP_RESEND_OTP_TIMEOUT_IN_SEC="120"
-REACT_APP_SEND_OTP_CHANNELS="email,mobile"
\ No newline at end of file
+REACT_APP_SEND_OTP_CHANNELS="email,mobile"
+REACT_APP_SEND_OTP_SHOW_CAPTCHA="false"
```
Comment (gsasikumar) [inline .env:31]: why is this "SEND_OTP_SHOW_CAPTCHA"? why can't we just call it as just SHOW_CAPTCHA

File: package.json [hunk 5]
```diff
@@ -16,6 +16,7 @@
     "qrcode": "^1.5.1",
     "react": "^18.2.0",
     "react-dom": "^18.0.0",
+    "react-google-recaptcha": "^2.1.0",
```
Comment (gsasikumar) [inline package.json:19]: This should be added only if needed. Is it possible? if not ok.

**Review by gsasikumar — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (anshulv1401) [reply .env:33]: "6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI" is a dummy key. It shows like this
image

Comment (gsasikumar) [reply .env:33]: grt

File: src/services/authService.js (hunk 3 above)
Comment (anshulv1401) [reply src/services/authService.js:156]: then null will be passed, and the request body will not be affected.

File: .env (hunk 4 above)
Comment (anshulv1401) [reply .env:31]: SEND_OTP suggests the component these env variables are used in. can be removed if required.

File: src/services/authService.js (hunk 3 above)
Comment (gsasikumar) [reply src/services/authService.js:156]: ok grt

**Review by gsasikumar — APPROVED** [review]

File: public/locales/en.json [hunk 6]
```diff
@@ -90,20 +94,23 @@
     "OTP": "OTP"
   },
   "idpDetails": {
-    "idp_details_heading": "IDP Details",
+    "idp_details_heading": "OIDC Details",
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply public/locales/en.json]: IDP Details will updated as  Esignet well-known APIs

File: public/locales/en.json [hunk 7]
```diff
@@ -90,20 +94,23 @@
...
     "internal_api": "Internal API",
     "wellknown_api": ".wellknown API"
   },
   "header": {
-    "login_heading": "Login with MOSIP",
-    "login_paragraph": "Have not registered for MOSIP yet?",
+    "login_heading": "Login with e-Signet",
+    "login_paragraph": "Have not registered for e-Signet yet?",
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply public/locales/en.json]: login_paragraph is not used in the code, hence removed

File: src/components/L1Biometrics.js [hunk 8]
```diff
@@ -127,9 +129,9 @@ export default function L1Biometrics({
 
     try {
       await Authenticate(
-        getTransactionId(),
+        oAuthDetailsService.getTransactionId(),
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply src/components/L1Biometrics.js]: renamed oAuthDetailsService as openIDConnectService

File: src/components/OtpVerify.js [hunk 9]
```diff
@@ -189,8 +187,24 @@ export default function OtpVerify({
...
+
+        let params = "?";
+        if (nonce) {
+          params = params + "nonce=" + nonce + "&";
+        }
+        if (state) {
+          params = params + "state=" + state + "&";
+        }
+
+        let responseB64 = oAuthDetailsService.encodeBase64(oAuthDetailsService.getOuthDetails());
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply src/components/OtpVerify.js]: name corrected to getOAuthDetails()

File: src/pages/Login.js [hunk 10]
```diff
@@ -4,18 +4,20 @@ import Pin from "../components/Pin";
...
+import sbiService from "../services/sbiService";
 import Background from "../components/Background";
 import SignInOptions from "../components/SignInOptions";
 import {
   configurationKeys,
   validAuthFactors,
 } from "../constants/clientConstants";
-import { linkAuthService } from "../services/linkAuthService";
+import linkAuthService from "../services/linkAuthService";
 import IDPQRCode from "../components/IDPQRCode";
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply src/pages/Login.js]: IDPQRCode name updated to LoginQRCode

File: .env (hunk 4 above)
Comment (anshulv1401) [reply .env:31]: SEND_OTP_SHOW_CAPTCHA renamed to REACT_APP_CAPTCHA_ENABLE which will be a comma-separated list of components where the captcha needs to be shown

File: package.json (hunk 5 above)
Comment (anshulv1401) [reply package.json:19]: we are dependent on the OAuth response to know if we want to show a captcha or not. Hence cannot determine the library's requirement at the build time.

**Pull request merged**